content/learning-paths/servers-and-cloud-computing/cca-container/hello-realm.md

```diff
@@ -0,0 +1,132 @@
+{
+  printf("Hello from the Realm!\n");
+  sleep(99);
+}
+```
+
+Now compile the application as the init executable and package it into the `initramfs`:
+
+```console
+aarch64-linux-gnu-gcc -static hello.c -o init
+echo init | cpio -o -H newc | gzip > test.cpio.gz
+```
+`test.cpio.gz` is the compressed archive you will use as the initramfs for the guest linux kernel.
+
+## Run the application inside the Realm 
+
+Start the docker container with the `FM-share` directory mounted:
+
+```console
+docker run -v ~/FM-share:/home/ubuntu/FM-share -it armswdev/aemfvp-cca-image /bin/bash
+```
+Run the Arm CCA reference stack pre-built binaries on the FVP:
+
+```console
+./run-cca-fvp.sh
+```
+After the host linux kernel has booted up, you will prompted to login. Use `root` for both the login and password:
+
+```output
+[    3.879183] VFS: Mounted root (ext4 filesystem) readonly on device 254:0.
+[    3.879643] devtmpfs: mounted
+[    3.897918] Freeing unused kernel memory: 9024K
+[    3.901685] Run /sbin/init as init process
+[    3.965637] EXT4-fs (vda): re-mounted 34fdcd00-4506-42fc-8af8-a35ca9cf965d r/w. Quota mode: none.
+Starting syslogd: OK
+Starting klogd: OK
+Running sysctl: OK
+Saving random seed: OK
+Starting network: ip: RTNETLINK answers: File exists
+udhcpc: started, v1.31.1
+udhcpc: sending discover
+udhcpc: sending select for 172.20.51.1
+udhcpc: lease of 172.20.51.1 obtained, lease time 86400
+deleting routers
+adding dns 172.20.51.254
+FAIL
+Starting dropbear sshd: OK
+
+Welcome to Buildroot
+buildroot login: root
+Password:
+#
+```
+With the host linux kernel running, you can now enable sharing of files between your host machine and the running simulation. Mount a `9p` drive on the host linux kernel using `/etc/fstab`:
+
+```console
+vi /etc/fstab
+```
+In the `/etc/fstab` file, add the following line to it:
+
+```console
+FM	/mnt		9p	trans=virtio,version=9p2000.L,aname=/mnt	0 	0"
+```
+Save and close the file. Now run the mount command:
+
+```console
+mount -a
+```
+This command mounts the `FM-share` directory in your docker container to the `/mnt` directory of the host linux kernel running in the FVP. Inspect the contents of `/mnt`:
+
+```console
+ls /mnt
+```
+
+The output should be:
+```output
+hello.c  init  test.cpio.gz
+```
+You can now use `kvmtool` which is a virtualization utility to launch linux guest images. It supports the creation of realm guests. Use the `-i` switch to point to the `initramfs` root filesystem you created in the previous step.
+
+```console
+lkvm run --realm --irqchip=gicv3-its -p earlycon --network mode=none -c 1 -m 128 -k /realm/Image -i /mnt/test.cpio.gz
+```
+The realm guest takes some time to boot up. At the end of the boot, you should see your hello application running in the Realm:
+
+```output
+[    9.721618] Freeing unused kernel memory: 9024K
+[    9.740198] Run /init as init process
+Hello from the Realm!
+```
+You have successfully run an application inside the Realm!
+
```

Comment:
Again, I would double down here with another reminder that the whole realm VM will exit after the application runs, because it has been packaged as the kernel's `init` process.

Reply:
Done.